Add scripts for detailed comparison of two offline search runs

josh-willis opened the pull request:
This PR adds to the tools directory some scripts for a detailed comparison between two runs of the offline search. This is intended for validation of changes, and so far is focused on the case where the improvement is computational and the two runs would ideally have similar sensitivity. These scripts are only aware of the two IFO search configuration and would need adaptation to the multi-IFO case.

GarethDaviesGW commented:
For info: there is already a vt ratio plotting script available in [`bin/plotting/pycbc_plot_vt_ratio`](https://github.com/gwastro/pycbc/blob/master/bin/plotting/pycbc_plot_vt_ratio)

It may be best to either remove that one, move that here in place of the new one or merge the two in terms of functionality

spxiwh commented:
@GarethDaviesGW Can you comment on which of the two scripts is more functional?

@josh-willis This feels more like new executables than tools (tools doesn't get looked at much). Is there any reason not to put these in the `bin` directory, in some appropriate (perhaps new) sub-directory?

GarethDaviesGW commented:
> @GarethDaviesGW Can you comment on which of the two scripts is more functional?

They're actually fairly different looking at the outputs ([new](https://wiki.ligo.org/pub/CBC/Searches/PyCBC/WaveformCompressionAndGPUReview/cpu_compressed_vs_uncompressed_VT.png) and [previous](https://user-images.githubusercontent.com/44500294/82683447-59236f80-9c51-11ea-8b6d-3b1f6bc60c07.png)) - the previous one selects which IFAR to use and plots an error bar at each value, but I think they're both useful for different things. I think that the new code is probably more useful as a test output (more detail), but the previous code is more useful for presentation of results, as I think that it is easier to understand and better to show significant difference

I would argue to keep both, but what each one should be called in order to separate them I'm not sure.

I just wanted to bring attention to the existence of a similar code, and hadn't realised the extent of the differences between them

josh-willis commented:
@GarethDaviesGW Thanks for pointing out your code.  In fact, after you did so I remembered that I had found that script, and had hoped it would plot what I needed, but realized it did not.  I've renamed my script `pycbc_plot_vt_ratio_vs_ifar` to distinguish it from yours, and in the process of adding copyright statements to all of the scripts in this commit, I also added one to that script that credits you, since I'm pretty sure I borrowed from your script (at least the propagation of VT ratio uncertainties).

@spxiwh As Gareth notes, the scripts do somewhat different things.  What I've added (and now renamed) basically plots the ratio of our standard VT plot on the summary page.  Since the usual plot is on a log-log scale, it can be hard to judge the significance of small changes visually.  Also, the beginning of the waveform compression review had specifically asked for this, so I needed to do it anyway.  But I do think it is generally useful when comparing two runs, to see how VT might change as a function of both mass parameter space and IFAR.

I've also moved all of the scripts to a sub-subdirectory of `bin/`, with hopefully clearer names. I've made the directory structure so that if in the future, we find workflow comparisons between other workflows than the offline search to be useful, it is not hard to create a new sub-directory to hold those as well.

Hopefully these changes address the concerns raised in the comments above.
